### PR TITLE
Add support for JetStream publishing

### LINKED ISSUE #215: Publish to jetstream with ack
#### Use Case:
Hopefully I didn't missed that, but I don't see an option for having ack from jetstream when publishing messages.
That should be in place to fully utilize jetstreams.

#### Proposed Change:
Have a Publish implementation in nats::jetstream module That sends request to jetstream subject and process the response.

#### Who Benefits From The Change(s)?
Everybody using jetstream.

#### Alternative Approaches
Publish without waiting for ack.

### PR DESCRIPTION
**Pull request by caspervonb**
This adds adds support for `JetStream` publishing.
A few tangental fixes were made along the way.

Supercedes #218 
Closes #215

### CODE REVIEW

File: src/jetstream.rs [hunk 1]
```diff
@@ -603,6 +608,124 @@ impl JetStream {
...
+        subject: &str,
+        data: impl AsRef<[u8]>,
+        options: &PublishOptions,
+    ) -> io::Result<PublishAck> {
+        self.publish_with_options_or_headers(subject, Some(options), None, data)
+    }
+
+    /// Publishes a `Message` to `JetStream`.
+    pub fn publish_message(&self, message: &Message) -> io::Result<PublishAck> {
+        self.publish_with_options_or_headers(&message.subject, None, None, &message.data)
```
Comment (Jarema) [inline src/jetstream.rs:628]: Message may contain Headers, yet we ignore them here.

File: src/jetstream.rs [hunk 2]
```diff
@@ -603,7 +608,125 @@ impl JetStream {
...
```
Comment (Jarema) [inline src/jetstream.rs:628]: Message may contain Headers, yet we ignore them here.

**Review by Jarema — CHANGES_REQUESTED** [review]

File: src/jetstream.rs (hunk 2 above)
Comment (caspervonb) [reply src/jetstream.rs:628]: Yikes, good catch.

**Review by Jarema — APPROVED** [review]

**Comment by Jarema** [comment]
LGTM

**Pull request merged**